pkg/credentials/https/https_secret.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package https
+
+import (
+	"github.com/kubeflow/kfserving/pkg/constants"
+	v1 "k8s.io/api/core/v1"
+	"strings"
+)
+
+// Create constants -- baseURI
+const (
+	HTTPSHostURI = "https-base-uri"
+	HEADERS      = "headers"
+	HEADER       = "header"
+	NEWLINE      = "\n"
+)
+
+var (
+	InferenceServiceHTTPSHostURI = constants.KFServingAPIGroupName + "/" + HTTPSHostURI
+	HeaderPrefix                 = HEADER + "."
+	CommaSeparator               = ","
+	ColonSeparator               = ": "
+)
+
+// Can be used for http and https uris
+func BuildSecretEnvs(secret *v1.Secret) []v1.EnvVar {
+	var fieldKeys []string
+	envs := []v1.EnvVar{}
+	hostURI, ok := secret.Annotations[InferenceServiceHTTPSHostURI]
```

Comment:
wdyt about using secret type VS annotation ? https://kubernetes.io/docs/concepts/configuration/secret/#secret-types

Reply:
I think using secret's data type may be better in this case for consistency with the [other conditions](https://github.com/kubeflow/kfserving/blob/master/pkg/credentials/service_account_credentials.go#L98). Provides more confidentiality I suppose.